sunpy/database/database.py

```diff
@@ -378,6 +378,20 @@ def _download_and_collect_entries(self, query_result, client=None,
         if client is None:
             client = VSOClient()
 
+        remove_list = []
+        for qr in query_result:
+            temp =  tables.DatabaseEntry._from_query_result_block(qr)
+            for database_entry in self:
+                if database_entry.path is not None and temp._compare_attributes(
+                    database_entry, ["source", "provider", "physobs", "fileid", 
+                    "observation_time_start", "observation_time_end", 
+                    "instrument", "size", "wavemin", "wavemax"]):
+                    remove_list.append(qr)
+                    break
+
+        for temp in remove_list:
```

Comment:
Can you not combine this loop with the previouprevious one? I.e. instead of creating remove_list just remove them from query_result?